### DISCUSSION TITLE
Update Adam learning rate after 10 epochs

### DISCUSSION THREAD

**Discussion by cmdrootaccess** [Lightning-AI/pytorch-lightning#9853] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule] [answered]
hello,

how can i update the learning rate of adam optimizer after 10 epochs ?
my code is like this
```
self.lr_decay_epoch = [15,]
if epoch in self.lr_decay_epoch:
     self.lr = self.lr * 0.1
     self.optimizer = Adam(filter(lambda p: p.requires_grad, self.net.parameters()), lr=self.lr, weight_decay=self.wd)
```

**Comment by rohitgr7** [comment, marked as answer]
you can use [LambdaLR](https://pytorch.org/docs/stable/generated/torch.optim.lr_scheduler.LambdaLR.html#) where lambda function can be something like:
```py
lambda epoch: return lr * (0.1 if epoch in self.lr_decay_epoch else 1)
```

**Reply by cmdrootaccess** [reply]
can you please show me that in pytorch lightning ?

**Reply by rohitgr7** [reply]
```py
def configure_optimizers(self):
    opt = # your optimizer
    sched = LambdaLR(opt, lr_lambda=lambda epoch: return lr * (0.1 if epoch in self.lr_decay_epoch else 1))
    return {'optimizer': opt, 'lr_scheduler': sched}
```

**Reply by cmdrootaccess** [reply]
thank you so much 😊
